api/v1alpha1/serverbios_types.go

```diff
@@ -0,0 +1,69 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and IronCore contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// ServerBIOSSpec defines the desired state of ServerBIOS
+type ServerBIOSSpec struct {
+	// ServerRef is a reference to Server object
+	// +optional
+	ServerRef v1.LocalObjectReference `json:"serverRef,omitempty"`
+
+	// BIOS contains a bios version and settings.
+	// +optional
+	BIOS BIOSSettings `json:"bios,omitempty"`
+}
+
+// BIOSSettings contains a version, settings and a flag defining whether it is a current version
+type BIOSSettings struct {
+	// Version contains BIOS version
+	// +required
+	Version string `json:"version"`
+
+	// Settings contains BIOS settings as map
+	// +optional
+	Settings map[string]string `json:"settings,omitempty"`
+}
+
+// ServerBIOSStatus defines the observed state of ServerBIOS
+type ServerBIOSStatus struct {
+	// BIOS contains a bios version and settings.
+	// +optional
+	BIOS BIOSSettings `json:"bios,omitempty"`
```

Comment:
@stefanhipfel Taking into account:
- possible separating of metal-operator and bios/fw operator
- possible separation of the bios/fw operator and task runner

I'd suggest to proceed as is for now. Then we could add some endpoint to retrieve active tasks for concrete server or something like this. But it will require to store the references for running tasks etc. So I'd avoid adding it right now due to necessity to design this better.